wirecard/gambio-ee
# WDG-10: Docker setup

### This PR

* Adds a Docker setup that contains a stable Gambio version and the plugin ready to be activated
* Adds [Rake](https://github.com/ruby/rake) as a task runner for custom tasks, including:
  * `rake up`: Starts all Docker containers
  * `rake down`: Tears down all Docker containers
  * `rake bash`: Allows running commands from within the main container
  * `rake clear_cache`: Clears all shop caches

### Notes

* The installation comes with some default data and users; the default admin user is:
  User: `admin@admin.com`
  Password: `admin`

### How to Test

* Unless it's already installed on your system, [install Bundler](https://bundler.io/#getting-started)
* Run `bundler install` to install Rake
* Using the `.env.template` file, create an `.env` file
* Run `rake up` to start the Docker containers
* Wait for the containers to be up and the shop to be installed (this might take a few seconds)
* Open http://localhost/ and check if the shop works properly, e.g. articles can be ordered
* Open http://localhost/login_admin.php and login to the admin with the above credentials
* Navigate to the [Module Center](http://localhost/admin/admin.php?do=ModuleCenter) and confirm that the _Wirecard Gambio Module_ can be activated

### Jira Links

* WDG-10: https://jira.parkside.at/browse/WDG-10
